### PR TITLE
Implement AOS-62 WriteMultipart returns Part

### PR DESCRIPTION
**Pull request by JinnyYi**
ref: https://github.com/aos-dev/go-storage/issues/571.

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -508,18 +549,21 @@ func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int6
...
+		Bucket: &s.name,
+		// For S3, the `PartNumber` is [1, 10000]. But for users, the `PartNumber` is zero-based.
+		// Set PartNumber=index+1 here to ensure pass in an effective `PartNumber` for `UploadPart`.
+		// ref: https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html
+		PartNumber:    aws.Int64(int64(index + 1)),
 		Key:           aws.String(o.ID),
 		UploadId:      aws.String(o.MustGetMultipartID()),
 		ContentLength: &size,
-		Body:          iowrap.SizedReadSeekCloser(r, size),
+		Body:          aws.ReadSeekCloser(r),
```
Comment (Xuanwo) [inline storage.go:566]: Why we need to migrate from `iowrap.SizedReadSeekCloser` to `aws.ReadSeekCloser`?

Comment (JinnyYi) [reply storage.go:566]: I don't know why `WirteMultipart` got the ` RequestError` in the last commit. I found that `aws.ReadSeekCloser` is used in `write`, so I migrated to `aws.ReadSeekCloser` to have a try. The error info is:
```
RequestError: send request failed
...
net/http: HTTP/1.x transport connection broken: http: ContentLength=1543539 with Body length 0'
```

Comment (Xuanwo) [reply storage.go:566]: Please figure why.

If there is a bug in iowrap, please fix it. Don't    solve problem by skip it.

Comment (Xuanwo) [reply storage.go:566]: Maybe something buggy in https://github.com/aos-dev/go-storage/blob/master/pkg/iowrap/reader.go#L93-L108 ?

Comment (Xuanwo) [reply storage.go:566]: S3 SDK will compute Content MD5 by default which will consume our input reader.

In go-storage, we will let users calculate content md5 and passed into as a pair `Content-MD5`.

So we need to set `S3DisableContentMD5Validation` to true:

```go
cfg.S3DisableContentMD5Validation = aws.Bool(true)
```

Comment (Xuanwo) [reply storage.go:566]: Please change it back, we will use our own `iowrap` instead of the one provided by SDK.

**Pull request merged**